pkg/types/vsphere/validation/platform.go

```diff
@@ -94,3 +100,21 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateClient validates vCenter to only contain hostname and port
+func ValidateClient(vCenter string) error {
+	vCenterURL, err := url.Parse(vCenter)
+	if err != nil {
+		return err
+	}
+	if vCenterURL.Scheme != "" {
+		return errors.New("vCenter hostname should not contain url scheme")
+	}
+	if vCenterURL.Hostname() == "" {
```

Comment:
Did you test this out? It doesn't seem like `Hostname` is going to work: https://play.golang.org/p/0XjNkuIdP52

Abhinav's suggestion of https://golang.org/pkg/net/#SplitHostPort seems like a better fit.